src/psyclone/parse/algorithm.py

```diff
@@ -583,29 +660,68 @@ def get_kernel(parse_tree, alg_filename):
             # A simple variable e.g. arg
             full_text = str(argument).lower()
             var_name = full_text
-            arguments.append(Arg('variable', full_text, var_name))
+            try:
```

Comment:
You can do `datatype = arg_type_defns.get(var_name, None)` and then you don't need the `try...except`

Reply:
Good point. I dropped the optional None argument as it is the default.